discovery_engine_core/web-api/src/handlers.rs

```diff
@@ -60,10 +129,42 @@ pub(crate) async fn handle_user_interaction(
     }
 }
 
+/// Computes [`PositiveCoi`]s weights used to determine how many documents to fetch using each centers' embedding.
+fn compute_coi_weights(cois: &[PositiveCoi], horizon: Duration) -> Vec<f32> {
+    let relevances = compute_coi_relevances(cois, horizon, system_time_now());
+    let relevances = relevances
+        .iter()
```

Comment:
🔧 the second relevances computation could use the values instead of references
```suggestion
    let relevances = compute_coi_relevances(cois, horizon, system_time_now())
        .into_iter()
```